app/assets/javascripts/controllers.js

```diff
@@ -17,6 +17,27 @@ Prometheus.Graph = {
     's': 1
   },
 
+  durationToString: function(seconds) {
+    // TODO(stuart): this is always returning seconds
+    // look to round to a certain amount based on the units
+    // in the graph? some level of precision lost.
```

Comment:
the input is given in seconds as a large integer. Then, that number is `%`'d against the seconds value to see if it is perfectly divisible by it. If so, then the value returned will be in those units. otherwise, it will be in seconds.
example:

`durationToString(259200)` => '3d'
259200 % 60*60*24 (this is `factors['d']`) === 0, so it is evenly divisible by days

it is used at app/assets/javascripts/angular/resources/shared_graph_behavior.js:86, seen in this PR.

The input comes from the rickshaw graph. The seconds value is the range in seconds highlighted by the user. Rickshaw manages X axis times behind the scenes in seconds.